operations/mimir/store-gateway.libsonnet

```diff
@@ -61,6 +61,8 @@
     ),
     // Dynamically set GOMEMLIMIT based on memory request.
     GOMEMLIMIT: std.toString(std.floor($.util.siToBytes($.store_gateway_container.resources.requests.memory))),
+
+    JAEGER_REPORTER_MAX_QUEUE_SIZE: '1000',
```

Comment:
Do we want to make this a configuration setting in jsonnet for parity with Helm?

Reply:
Marco suggested not doing that in https://github.com/grafana/mimir/pull/6764/files#diff-f3070297971f928d50b986208d8f23344d5c0a1d37f2c1422eed86feea6bb31a, so I'm following the pattern established there. I'd be happy either way.

I went for a different approach in Helm because I believe it's a harder (or even impossible) to extend the default value of something, so it'd be tricky for someone to extend our default set of environment variables that includes `JAEGER_REPORTER_MAX_QUEUE_SIZE` with additional variables. Might be wrong though.